examples/ramen_rec/tests/test_constraint_solving.py

```diff
@@ -0,0 +1,54 @@
+from frex.pipeline_stages.scorers import CandidateRanker
+from examples.ramen_rec.app.models import RamenContext, RamenEaterContext
+from examples.ramen_rec.app.pipelines import *
+from examples.ramen_rec.tests.conftest import placeholder_ramen_candidate
+from frex.utils import ConstraintSolverUtils
+
+
+def test_choose_ramens_using_constraints(
+    sim_ramen_pipe,
+    test_ramen_101,
+):
+    output_candidates = tuple(
+        sim_ramen_pipe(
+            context=RamenContext(target_ramen=test_ramen_101),
+        )
+    )
+
+    csu = ConstraintSolverUtils()
+
+    final_candidates = csu.solve_constraints_on_candidates(
```

Comment:
If you end up using this code "for real" I'd like to see a nicer interface here. See comments below.